### ISSUE TITLE
submariner-networkplugin-syncer fails to start

### ISSUE THREAD

**Issue by aswinsuryan** [submariner-io/submariner#1878]
**What happened**:

When submariner is deployed with a cluster with OVNKubernetes CNI, submariner-networkplugin-syncer fails to start with this error.
`
F0620 21:36:09.217221       1 main.go:67] Error registering the handlers: Event handler "ovn-sync-handler" failed to initialize: error creating NBDB connection: error connecting to ovsdb: failed to connect to ssl:ovnkube-db.openshift-ovn-kubernetes.svc.cluster.local:9641: endpoint is not leader`

It starts working when the first IP  dig @<opnshift-dns-ip> ovnkube-db.openshift-ovn-kubernetes.svc.cluster.local
of the headless service is the current kuberntes master.

For example assume 192.168.198.14 is the current master. If the dig output is as below it does not work.
```
$dig @172.90.0.10 ovnkube-db.openshift-ovn-kubernetes.svc.cluster.local
ovnkube-db.openshift-ovn-kubernetes.svc.cluster.local. 5 IN A 192.168.198.239
ovnkube-db.openshift-ovn-kubernetes.svc.cluster.local. 5 IN A 192.168.198.14
ovnkube-db.openshift-ovn-kubernetes.svc.cluster.local. 5 IN A 192.168.198.181
```

Now if we delete pods with IP 192.168.198.239(one of ovnkube-master pod in openshift-ovn-kubernetes namespace)  . The output of dig changes as below and it starts working

```
$dig @172.90.0.10 ovnkube-db.openshift-ovn-kubernetes.svc.cluster.local
ovnkube-db.openshift-ovn-kubernetes.svc.cluster.local. 5 IN A 192.168.198.14
ovnkube-db.openshift-ovn-kubernetes.svc.cluster.local. 5 IN A 192.168.198.181
ovnkube-db.openshift-ovn-kubernetes.svc.cluster.local. 5 IN A 192.168.198.239

```

In short, It starts working when submariner-networkplugin-syncer connects to correct leader.

**What you expected to happen**:

The submariner-networkplugin-syncer pod should be up and running

**How to reproduce it (as minimally and precisely as possible)**:
Deploy a Openshift 4.11 cluster and deploy submariner. If it is working try deleting the current ovn master and submariner-networkplugin-syncer.

**Anything else we need to know?**:

**Environment**:
- Diagnose information (use `subctl diagnose all`):
- Gather information (use `subctl gather`):
- Cloud provider or hardware configuration:
- Install tools:
- Others:

**Comment by nyechiel** [comment]
@sridhargaddam @vthapar @astoycos have you seen this behavior before? Seems like something we need to address in 0.13

**Comment by astoycos** [comment]
Off the top of my head if we initialize the clients with https://github.com/submariner-io/submariner/blob/devel/pkg/networkplugin-syncer/handlers/ovn/connection.go#L127 set to `false` this error _should_ go away, and it may simply make sense to do this since monitoring for leadership changes could be a bit costly/ all db changes should be mirrored throughout the cluster/ Submariner isn't a "write heavy" client at all. 

However, I would like to confer with the OVN/OVN-K teams to better understand if we should only be communicating with the active master dbs here which I will do first thing tomorrow once folks are back online.
